PWGHF/TableProducer/HFXiccToPKPiPiCandidateSelector.cxx

```diff
@@ -68,8 +68,23 @@ struct HfXiccToPKPiPiCandidateSelector {
       return false;
     }
 
-    // candidate decay length
-    if (hfCandXicc.decayLength() <= cuts->get(pTBin, "decay length")) {
+    // cosine of pointing angle XY
+    if (hfCandXicc.cpaXY() <= cuts->get(pTBin, "cos pointing angle XY")) {
+      return false;
+    }
+
+    // candidate maximum decay length
+    if (hfCandXicc.decayLength() > cuts->get(pTBin, "max decay length")) {
+      return false;
+    }
+
+    // candidate maximum decay length XY
+    if (hfCandXicc.decayLength() > cuts->get(pTBin, "max decay length XY")) {
+      return false;
+    }
+
+    // candidate minimum decay length
+    if (hfCandXicc.decayLength() <= cuts->get(pTBin, "min decay length")) {
```

Comment:
Is it intentional to accept the maxima and reject the minima?

Reply:
The statement 
`if (hfCandXicc.decayLength() > cuts->get(pTBin, "max decay length XY")) {
      return false;
    }`
is also rejecting candidates with decay length larger than maximum. Where do you see that the maxima are accepted?

I also found an error in this if-statement. The cut is applied on `max decay length XY` but the variable used in the if-condition is `hfCandXicc.decayLength()`. I fixed it with a commit

Reply:
`if (hfCandXicc.decayLength() == cuts->get(pTBin, "max decay length XY"))` you accept the candidate, isn't it?

Reply:
No, because the condition inside the if-statement is `return false`

Reply:
But the statement is not executed in that case. `return false` is executed only if `>`, not if `==`.

Reply:
No, the if condition is always checked. If the condition is satisfied (i.e. the `hfCandXicc.decayLength` is `<= cuts->get(pTBin, "max decay length")` it returns `false`. This means that the candidate has a decayLength lower than the cut and therefore is not selected (i.e. `return false`)